server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3520,27 +3520,17 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                     }
                     rootDiskSize = Long.parseLong(customParameters.get("rootdisksize"));
 
-                    // only KVM supports rootdisksize override
-                    if (hypervisorType != HypervisorType.KVM) {
-                        throw new InvalidParameterValueException("Hypervisor " + hypervisorType + " does not support rootdisksize override");
+                    // only KVM, XenServer and VMware  supports rootdisksize override
+                    if (!(hypervisorType == HypervisorType.KVM || hypervisorType == HypervisorType.XenServer || hypervisorType == HypervisorType.VMware)) {
+                        throw new InvalidParameterValueException("Hypervisor " + hypervisorType + " does not support  rootdisksize override");
```

Comment:
The condition in VolumeApiServiceImpl.java at line# 1029 above and this doesn't match. You skipped None and Any. Is that intentional?